src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= b + c || a <= Math.abs(b - c)) {
+      throw new Error('Given sides of the triangle can\'t form a triangle');
+    }
 
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        'One or more sides of the triangle are less or equal zero',
+      );
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter: number = (this.a + this.b + this.c) / 2;
+    const area = Math.floor(
+      Math.sqrt(semiPerimeter
+        * (semiPerimeter - this.a)
+        * (semiPerimeter - this.b)
+        * (semiPerimeter - this.c))
+      * 100,
+    ) / 100;
+
+    return area;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius of the circle is less or equal zero');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(100 * Math.PI * (this.radius ** 2)) / 100;
```

Comment:
Create variable for this `Math.PI * (this.radius ** 2)`
Also, remember about DRY. You are using next logic at least twice in your code `Math.floor(100 * ...) / 100;`
Create function for this